sympy/solvers/solvers.py

```diff
@@ -1557,7 +1557,10 @@ def _expand(p):
                         try:
                             t = Dummy('t')
                             iv = _solve(u - t, symbol, **flags)
-                            soln = list(ordered({i.subs(t, s) for i in iv for s in soln}))
+                            new_soln = list(ordered({i.subs(t, s) for i in iv for s in soln}))
```

Comment:
Let's take a slightly more complicated expression: `x**(2/x) + 4*x**(1/x) +3`. What `_solve` tries to do is to go like, "Okay, can I make a polynomial out of this?" And yes, it can: `t**2+4*t+3` where `t==x**(1/x)`. So now it can solve the polynomial first which gives `t=1` and `t=3`. So now you want to solve `x**(1/x)=1` and `x**(1/x)=3`. Solve does this in general: it solves `x**(1/x) = t` for a general t, and then substitutes the appropriate value to get the solutions of the original equation.